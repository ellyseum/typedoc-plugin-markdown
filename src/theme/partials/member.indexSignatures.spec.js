describe(`[partial] member.indexSignatures`, () => {
  test('should compile', () => {
    const context = {
      indexSignature: {
        name: '__index',
        flags: [],
        id: 534,
        parent: null,
        originalName: '__index',
        kind: 8192,
        parameters: [
          {
            name: 'index',
            flags: [],
            id: 535,
            parent: null,
            originalName: 'index',
            kind: 32768,
            type: { type: 'intrinsic', name: 'number' },
            kindString: 'Parameter',
          },
        ],
        type: { type: 'intrinsic', name: 'string' },
        sources: [
          {
            file: null,
            fileName: 'interfaces.ts',
            line: 70,
            character: 32,
            url:
              'https://github.com/tgreyuk/typedoc-plugin-markdown/blob/master/test/src/interfaces.ts#L70',
          },
        ],
        kindString: 'Index signature',
      },
    };
    const result = global.compileTemplate('member.indexSignatures.hbs', context);
    expect(result).toMatchSnapshot();
  });
});
