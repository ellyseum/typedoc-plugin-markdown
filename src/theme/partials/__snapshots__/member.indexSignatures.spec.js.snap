// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[partial] member.indexSignatures should compile 1`] = `
"[index: [type]]:

[type]

[comment]"
`;
